### PR TITLE
Storage upload updates

### PR DESCRIPTION
**Pull request by cicdw**
This PR tidies up some issues with blocks related to deployments:
- stops redundantly saving storage blocks as anonymous blocks
- stops saving default infra blocks on agents as anonymous blocks
- exposes a `--skip-upload` flag on `build` to avoid remote storage uploads (related to https://github.com/PrefectHQ/prefect/issues/6376 but does not take it that far yet)

Example usage:
```
prefect deployment build ./flows/example.py:main --name "Example" -sb s3/dev --skip-upload
# user is now responsible for performing the file upload
prefect deployment apply main-deployment.yaml
```

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a `bug`, `feature`, `enhancement`, `docs`, or `maintenance` label categorizing the change.

### CODE REVIEW

File: src/prefect/agent.py [hunk 1]
```diff
@@ -262,12 +262,6 @@ async def start(self):
         await self.client.__aenter__()
         await self.task_group.__aenter__()
 
-        # Convert the passed default infrastructure to an id
-        if self.default_infrastructure and not self.default_infrastructure_document_id:
-            self.default_infrastructure_document_id = (
-                await self.default_infrastructure._save(is_anonymous=True)
-            )
-
```
Comment (madkinsz) [inline src/prefect/agent.py:270]: With this change we'll fail at https://github.com/PrefectHQ/prefect/blob/33350dfb901a994000739bdf5f13338a0c3c1c55/src/prefect/agent.py#L161-L165

**Review by anna-geller — APPROVED** [review]
QA passed - deployments that share the same -ib but with overrides now point to the same block without creating an anonymous block

```
prefect deployment build flows/healthcheck.py:healthcheck --name process -q prod -t project \
-o deploy/process.yaml -ib process/prod -v GITHUB_SHA
prefect deployment apply deploy/process.yaml
prefect deployment run healthcheck/process

prefect deployment build flows/healthcheck.py:healthcheck --name process2 -q prod -t project \
-o deploy/process2.yaml -ib process/prod --override env.PREFECT_LOGGING_LEVEL=DEBUG -v GITHUB_SHA
prefect deployment apply deploy/process2.yaml
prefect deployment run healthcheck/process2
```
image

agent picked up those runs correctly

skip upload on storage blocks also works
image

**Comment by madkinsz** [comment]
I noticed this includes a new `--skip-upload` flag, can you include an example of how that's used in the pull request body?

File: src/prefect/agent.py (hunk 1 above)
Comment (cicdw) [reply src/prefect/agent.py:270]: @madkinsz updated 👍 good catch

**Review by madkinsz — APPROVED** [review]

**Pull request merged**